File: Python/02_Can Make Arithmetic Progression From Sequence.py
# 1502. Can Make Arithmetic Progression From Sequence
# https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/

class Solution:
    def canMakeArithmeticProgression(self, arr):
        arr.sort()
        x=arr[1]-arr[0]
        for i in range(2,len(arr)):
            if arr[i]-arr[i-1]!=x:
                return False
        return True

File: Java/02_can_make_arithmetic_progression_from_sequence.java
/*
 * 1502. Build Array from Permutation
 * https://leetcode.com/problems/can-make-arithmetic-progression-from-sequence/
 */


class Solution {
    public boolean canMakeArithmeticProgression(int[] arr) {
        Arrays.sort(arr);
        int dif=arr[1] - arr[0];
        for(int i=2; i<arr.length; ++i){
            if(arr[i]-arr[i-1] != dif){
                return false;
            }
        }
        return true;
    }
}
